src/move.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+var Move = function(state) {
+  this.state = state;
+  this.lastPosition = {};
```

Comment:
When moving this to the selection object, you can have a local variable by using `var myVariable` inside the function. Javascript has function scope (a function declaration creates a closure), so as long as you use the `var` keyword you can access to a `lastPosition` variable within that closure without polluting the scope of the selection object if you need it.